beater/beater.go

```diff
@@ -237,6 +240,7 @@ func (bt *beater) start(ctx context.Context, cancelContext context.CancelFunc, b
 type serverCreator struct {
 	context context.Context
 	args    sharedServerRunnerParams
+	runner  *serverRunner
```

Comment:
The serverCreator holds a reference to the runner; when it receives a new config it parses that (as it does now), but instead of creating a new runner it will pass the params so it can update it's configuration.